primefaces/src/main/java/org/primefaces/component/barcode/BarcodeHandler.java

```diff
@@ -85,13 +94,20 @@ public void handle(FacesContext context) throws IOException {
         }
 
         try {
-            BarcodeGenerator generator = generators.get(params.get("gen"));
-            String format = params.get("fmt");
+            String type = params.get("gen").toLowerCase(Locale.ROOT);
+            Symbol generator = GENERATORS.get(type).get();
+            String format = params.get("fmt").toLowerCase(Locale.ROOT);
             String hrp = params.get("hrp");
             int orientation = Integer.parseInt(params.get("ori"));
+            double magnification = Double.parseDouble(params.get("mag"));
             boolean cache = Boolean.parseBoolean(params.get(Constants.DYNAMIC_CONTENT_CACHE_PARAM));
 
-            generator.getBarcodeBean().setMsgPosition(HumanReadablePlacement.byName(hrp));
+            generator.setHumanReadableLocation(HumanReadableLocation.valueOf(hrp.toUpperCase(Locale.ROOT)));
+            generator.setContent(value);
+
+            if ("qr".equals(type)) {
```

Comment:
To avoid unsafe cast warning, it might be "better" to use `instanceof QrCode`

Reply:
Done!